src/components/Postiterie.vue

```diff
@@ -1,203 +1,454 @@
     </div>
-    <div id="result-hr" v-bind:class="{hidden: isNoResult}">
-      <div class="hr">
-        <span class="hr-img">.</span>
-        <span>Résultat</span>
-        <span class="hr-img">.</span>
+
+    <div class="upload-container" v-if="authorized && image && !link">
+      <div class="input">
+        <input placeholder="nom du fichier" type="text" v-model="fileName">
       </div>
+      <div class="round-button_container" >
+        <button @click="tracking" class="upload button"><i class="material-icons">cloud_upload</i></button>
+      </div>             
+    </div> 
+
+    <div class="info" v-else-if="authorized && !link">
+      Vous êtes authentifié à google drive !<br> Prenez une photo ou importez en une depuis votre galerie pour générer un schéma
+    </div>
+
+    <div class="upload-container" v-else-if="!link">
+        <button id="authorize_button" class="button button-orange">S'authentifier</button>        
     </div>
-    <div id="result">
-        <li v-for="image in results">
-          <img v-bind:src='image.currentSrc' >
-        </li>
+
+    <div class="button-container" v-if="link">
+      <a class="button button-orange" target="_blank"  v-bind:href="link">Ouvrir le schéma avec drawIO</a> 
     </div>
+    
+    <!-- has to be here for tracking-->
+    <img style="display:none"  id="preview" :src="image" alt="">
   </div>
+
 </template>
 <script src="../src/tracking.js"></script>
 
-
 <script>
 
-/* 
-  FUNCTION imageDataToImage
-  transform an imageData to an image
-  IN : imageData
-  OUT : image
-*/
-function imageDataToImage(imagedata) {
-    let canvas = document.createElement('canvas');
-    let ctx = canvas.getContext('2d');
-    canvas.width = imagedata.width;
-    canvas.height = imagedata.height;
-    ctx.putImageData(imagedata, 0, 0);
-    let image = new Image();
-    image.src = canvas.toDataURL();
-    return image;
-}
-
-/*
-  FUNCTION crop
-  Crop an image using the coordinates and return its imageData
-  IN : x, y (numbers) coordinates from which to start croping
-       width, height (numbers) length to crop
-  OUT: imageData
-*/
-function crop(x, y, width, height){
-  let canvas = document.createElement('canvas');
-  let context = canvas.getContext('2d');
-  let img = document.getElementById('preview');
-  canvas.width = img.width;
-  canvas.height = img.height;
-  context.drawImage(img, 0, 0);
-  return context.getImageData(x, y, width, height);
-}
+  var authorizeButton;
 
 export default {
+
   name: 'Postiterie',
   data () {
     return {
       image: null,
-      isNoImage: true,
-      isNoResult: true,
-      results: []
+      results: [],
+      authorized: false,
+      trackingFinished: false,
+      trackingFinishedWithNoResult: false,
+      fileName: "",
+      CLIENT_ID: '222333408772-7babrj6q1uronhdntc2a5fj6lv6m5cff.apps.googleusercontent.com',
```

Comment:
This should be in a configuration file. And the actual clientId and apiKey **shouldn't** be commited to github.